mne/report.py

```diff
@@ -949,46 +949,102 @@ def _validate_input(self, items, captions, section, comments=None):
 
         return items, captions, comments
 
-    def _add_figs_to_section(self, figs, captions, section='custom',
-                             image_format='png', scale=None, comments=None):
-        """Auxiliary method for `add_section` and `add_figs_to_section`."""
-        figs, captions, comments = self._validate_input(figs, captions,
-                                                        section, comments)
-        _check_scale(scale)
-        for fig, caption, comment in zip(figs, captions, comments):
-            caption = 'custom plot' if caption == '' else caption
-            sectionvar = self._sectionvars[section]
-            global_id = self._get_id()
-            div_klass = self._sectionvars[section]
-            img_klass = self._sectionvars[section]
+    def remove(self, fname=None, caption=None, section=None):
+        """Remove a figure from the report.
 
-            img = _fig_to_img(fig, image_format, scale)
-            html = image_template.substitute(img=img, id=global_id,
-                                             div_klass=div_klass,
-                                             img_klass=img_klass,
-                                             caption=caption,
-                                             show=True,
-                                             image_format=image_format,
-                                             comment=comment)
-            self.fnames.append('%s-#-%s-#-custom' % (caption, sectionvar))
-            self._sectionlabels.append(sectionvar)
+        The figures to remove can be searched for either by filename or by
+        their caption. The section label can be specified as well to narrow
+        down the search. If multiple figures match the search criteria, the
+        last one will be removed.
+
+        Parameters
+        ----------
+        fname : str | None
+            If set, search for the figure by filename.
+        caption : str | None
+            If set, search for the figure by caption.
+        section : str | None
+            If set, limit the search to the section with the given label.
+
+        Returns
+        -------
+        removed_fname : str | None
+            The filename of the figure that was removed, or ``None`` if no
+            figure matched the search criteria.
+        """
+        if (fname is None) == (caption is None):
+            raise ValueError('Specify either a filename or caption '
+                             '(but not both).')
+
+        # Construct the search pattern
+        if fname is not None:
+            pattern = r'^%s$' % fname
+        elif caption is not None:
+            pattern = r'^%s-#-.*-#-custom$' % caption
+
+        # Search for figures matching the search pattern, regardless of section
+        matches = [i for i, fname in enumerate(self.fnames)
```

Comment:
my flake8 editor complains here. Not sure why MNE does not pick it up:

"List comprehension redefines fname"